### DISCUSSION TITLE
How should I get JSON format path from expr inside leafref?

### DISCUSSION THREAD

**Discussion by lyzliyuzhi91** [CESNET/libyang#2066] [Q&A] [answered]
I get a leafref node, and I want to create an isntance for the leafref's target.
But once I call `lyd_new_path` by the path getted from `lyxp_get_expr`, it failed. It seems that because the path's prefix is not a module name but a module prefix.

**Comment by michalvasko** [comment, marked as answer]
You are right and there is no direct way of transforming the prefix into the module name so you will have to do so manually somehow.
